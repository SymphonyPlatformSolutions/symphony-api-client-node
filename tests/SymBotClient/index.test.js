const mockConfig = {
  agentHost: 'agent.example.com',
  agentPort: 443,
  podHost: 'pod.example.com',
  podPort: 443,
  getAgentHost: async () => 'agent.example.com',
  rotateAgent: async () => { }
}

jest.mock('../../lib/SymConfigLoader', () => ({
  SymConfig: mockConfig,
}))
jest.mock('../../lib/SymBotAuth', () => ({
  sessionAuthToken: 'session-token',
  kmAuthToken: 'key-manager-token',
}))
jest.mock('../../lib/DatafeedEventsService')

const https = require('https')
const nock = require('nock')
const SymBotClient = require('../../lib/SymBotClient')

describe('SymBotClient', () => {
  beforeAll(() => {
    jest.spyOn(https, 'request')
  })

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe.each([
    // AdminClient
    ['adminListEnterpriseStreamsV2', [], 'post', '/pod/v2/admin/streams/list?skip=0&limit=100'],
    [
      'streamMembers',
      ['abc123'],
      'get',
      '/pod/v1/admin/stream/abc123/membership/list?skip=0&limit=1000',
    ],
    ['importMessages', [['one', 'two']], 'post', '/agent/v4/message/import'],
    ['suppressMessage', ['abc123'], 'post', '/pod/v1/admin/messagesuppression/abc123/suppress'],

    // AdminUserClient
    ['getUser', ['abc123'], 'get', '/pod/v2/admin/user/abc123'],
    ['listUsers', [], 'get', '/pod/v2/admin/user/list?skip=0&limit=1000'],

    // ConnectionsClient
    ['getPendingConnections', [], 'get', '/pod/v1/connection/list?status=pending_outgoing'],
    ['getInboundPendingConnections', [], 'get', '/pod/v1/connection/list?status=pending_incoming'],
    ['getAllConnections', [], 'get', '/pod/v1/connection/list?status=all'],
    ['getAcceptedConnections', [], 'get', '/pod/v1/connection/list?status=accepted'],
    ['getRejectedConnections', [], 'get', '/pod/v1/connection/list?status=rejected'],
    [
      'getConnections',
      ['dummy', 'abc123'],
      'get',
      '/pod/v1/connection/list?status=all&userIds=abc123',
    ],
    ['acceptConnectionRequest', [], 'post', '/pod/v1/connection/accept'],
    ['rejectConnectionRequest', [], 'post', '/pod/v1/connection/reject'],
    ['sendConnectionRequest', [], 'post', '/pod/v1/connection/create'],
    ['removeConnection', ['abc123'], 'post', '/pod/v1/connection/user/abc123/remove'],
    ['getConnectionRequestStatus', ['abc123'], 'get', '/pod/v1/connection/user/abc123/info'],

    // MessagesClient

    // sendMessage - multipart boundary header
    // sendMessageWithAttachment - multipart boundary header
    // forwardMessage - multipart boundary header
    // getAttachment - response is Base64
    ['getMessage', ['abc123'], 'get', '/agent/v1/message/abc123'],

    // OBOClient

    // oboAuthenticateByUserId
    ['oboGetAllConnections', [], 'get', '/pod/v1/connection/list?status=all'],
    ['oboGetConnection', ['abc123'], 'get', '/pod/v1/connection/user/abc123/info'],
    ['oboGetUserIMStreamId', [], 'post', '/pod/v1/im/create'],
    // oboSendMessage - multipart boundary header

    // SignalsClient
    ['listSignals', [], 'get', '/agent/v1/signals/list?skip=0&limit=50'],
    ['getSignal', ['abc123'], 'get', '/agent/v1/signals/abc123/get'],
    ['createSignal', [], 'post', '/agent/v1/signals/create'],
    ['updateSignal', ['abc123'], 'post', '/agent/v1/signals/abc123/update'],
    ['deleteSignal', ['abc123'], 'post', '/agent/v1/signals/abc123/delete'],
    ['subscribeSignal', ['abc123'], 'post', '/agent/v1/signals/abc123/subscribe?pushed=false'],
    ['unsubscribeSignal', ['abc123'], 'post', '/agent/v1/signals/abc123/unsubscribe'],
    [
      'getSignalSubscribers',
      ['abc123'],
      'get',
      '/agent/v1/signals/abc123/subscribers?skip=0&limit=50',
    ],

    // StreamsClient
    ['getUserIMStreamId', [], 'post', '/pod/v1/im/create'],
    ['createRoom', [], 'post', '/pod/v3/room/create'],
    ['updateRoom', ['abc123'], 'post', '/pod/v3/room/abc123/update'],
    ['getRoomInfo', ['abc123'], 'get', '/pod/v3/room/abc123/info'],
    ['activateRoom', ['abc123'], 'post', '/pod/v1/room/abc123/setActive?active=true'],
    ['deactivateRoom', ['abc123'], 'post', '/pod/v1/room/abc123/setActive?active=false'],
    ['getRoomMembers', ['abc123'], 'get', '/pod/v2/room/abc123/membership/list'],
    ['addMemberToRoom', ['abc123'], 'post', '/pod/v1/room/abc123/membership/add'],
    ['removeMemberFromRoom', ['abc123'], 'post', '/pod/v1/room/abc123/membership/remove'],
    ['promoteUserToOwner', ['abc123'], 'post', '/pod/v1/room/abc123/membership/promoteOwner'],
    ['demoteUserFromOwner', ['abc123'], 'post', '/pod/v1/room/abc123/membership/demoteOwner'],
    ['searchRooms', [], 'post', '/pod/v3/room/search?skip=0&limit=100'],
    ['getUserStreams', [], 'post', '/pod/v1/streams/list?skip=0&limit=100'],

    // UsersClient
    ['getUserFromUsername', ['user123'], 'get', '/pod/v2/user?username=user123'],
    [
      'getUserFromEmail',
      ['example@example.com', true],
      'get',
      '/pod/v2/user?email=example@example.com&local=true',
    ],
    [
      'getUsersFromEmailList',
      ['example@example.com,example@example.org', true],
      'get',
      '/pod/v3/users?email=example@example.com,example@example.org&local=true',
    ],
    ['getUsersFromIdList', [1234, true], 'get', '/pod/v3/users?uid=1234&local=true'],
    [
      'searchUsers',
      ['query', true, 10, 100, 'filter'],
      'post',
      '/pod/v1/user/search?local=true&skip=10&limit=100',
    ],

    // PresenceClient
    ['getUserPresence', ['abc123'], 'get', '/pod/v3/user/abc123/presence?local=false'],
    ['setPresence', ['status'], 'post', '/pod/v2/user/presence'],
    ['registerInterestExtUser', [], 'post', '/pod/v1/user/presence/register'],
  ])('#%s', (methodName, args, verb, path) => {
    const host = path.substr(0, 6) === '/agent' ? mockConfig.agentHost : mockConfig.podHost

    it('handles success', async () => {
      nock(`https://${host}`)
      [verb](path)
        .reply(200, '{"status":"OK"}')

      await expect(SymBotClient[methodName](...args)).resolves.toEqual({
        status: 'OK',
      })

      expect(https.request.mock.calls).toMatchSnapshot()
    })

    it('handles failure', () => {
      nock(`https://${host}`)
      [verb](path)
        .replyWithError('Oh no')

      return expect(SymBotClient[methodName](...args)).rejects.toEqual({
        status: 'error',
      })
    })
  })

  describe('#getDatafeedEventsService', () => {
    it('supports old function signature', () => {
      const messageHandler = jest.fn()
      const feed = SymBotClient.getDatafeedEventsService(messageHandler)

      // call bound handler function
      feed.on.mock.calls[0][1]('hello')
      expect(messageHandler).toHaveBeenCalledWith('MESSAGE_RECEIVED', 'hello')

      expect(feed.start).toHaveBeenCalledWith(undefined)
    })

    it('supports new function signature', () => {
      const feedId = 'abc123'
      const onMessage = jest.fn()
      const onError = jest.fn()
      const feed = SymBotClient.getDatafeedEventsService({ onMessage, onError, feedId })

      expect(feed.start).toHaveBeenCalledWith(feedId)
      expect(feed.on).toHaveBeenCalledWith('message', onMessage)
      expect(feed.on).toHaveBeenCalledWith('error', onError)
    })

    it('allows errorHandler and feedId to be omitted', () => {
      const onMessage = jest.fn()
      const feed = SymBotClient.getDatafeedEventsService({ onMessage })

      expect(feed.start).toHaveBeenCalledWith(undefined)
      expect(feed.on).toHaveBeenCalledWith('message', onMessage)
      expect(feed.on).toHaveBeenCalledTimes(1)
    })
  })

  describe('#stopDatafeedEventsService', () => {
    it('stops instance', () => {
      jest.spyOn(console, 'warn').mockImplementationOnce(() => { })
      const feed = SymBotClient.getDatafeedEventsService(jest.fn())
      SymBotClient.stopDatafeedEventsService()
      expect(feed.stop).toHaveBeenCalled()
      expect(console.warn).toHaveBeenCalled()
    })
  })
})
