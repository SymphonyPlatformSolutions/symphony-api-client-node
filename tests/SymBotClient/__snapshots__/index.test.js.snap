// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SymBotClient #acceptConnectionRequest handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/connection/accept",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #activateRoom handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/room/abc123/setActive?active=true",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #addMemberToRoom handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/room/abc123/membership/add",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #adminListEnterpriseStreamsV2 handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "post",
      "path": "/pod/v2/admin/streams/list?skip=0&limit=100",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #createRoom handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v3/room/create",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #createSignal handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "keyManagerToken": "key-manager-token",
        "sessionToken": "session-token",
      },
      "hostname": "agent.example.com",
      "method": "POST",
      "path": "/agent/v1/signals/create",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #deactivateRoom handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/room/abc123/setActive?active=false",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #deleteSignal handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "keyManagerToken": "key-manager-token",
        "sessionToken": "session-token",
      },
      "hostname": "agent.example.com",
      "method": "POST",
      "path": "/agent/v1/signals/abc123/delete",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #demoteUserFromOwner handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/room/abc123/membership/demoteOwner",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getAcceptedConnections handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "GET",
      "path": "/pod/v1/connection/list?status=accepted",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getAllConnections handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "GET",
      "path": "/pod/v1/connection/list?status=all",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getConnectionRequestStatus handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "GET",
      "path": "/pod/v1/connection/user/abc123/info",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getConnections handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "GET",
      "path": "/pod/v1/connection/list?status=all&userIds=abc123",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getInboundPendingConnections handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "GET",
      "path": "/pod/v1/connection/list?status=pending_incoming",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getMessage handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "keyManagerToken": "key-manager-token",
        "sessionToken": "session-token",
      },
      "hostname": "agent.example.com",
      "method": "get",
      "path": "/agent/v1/message/abc123",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getPendingConnections handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "GET",
      "path": "/pod/v1/connection/list?status=pending_outgoing",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getRejectedConnections handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "GET",
      "path": "/pod/v1/connection/list?status=rejected",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getRoomInfo handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "GET",
      "path": "/pod/v3/room/abc123/info",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getRoomMembers handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "GET",
      "path": "/pod/v2/room/abc123/membership/list",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getSignal handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "keyManagerToken": "key-manager-token",
        "sessionToken": "session-token",
      },
      "hostname": "agent.example.com",
      "method": "GET",
      "path": "/agent/v1/signals/abc123/get",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getSignalSubscribers handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "keyManagerToken": "key-manager-token",
        "sessionToken": "session-token",
      },
      "hostname": "agent.example.com",
      "method": "GET",
      "path": "/agent/v1/signals/abc123/subscribers?skip=0&limit=50",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getUser handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "get",
      "path": "/pod/v2/admin/user/abc123",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getUserFromEmail handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "get",
      "path": "/pod/v2/user?email=example@example.com&local=true",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getUserFromUsername handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "get",
      "path": "/pod/v2/user?username=user123",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getUserIMStreamId handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/im/create",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getUserPresence handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "GET",
      "path": "/pod/v3/user/abc123/presence?local=false",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getUserStreams handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/streams/list?skip=0&limit=100",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getUsersFromEmailList handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "get",
      "path": "/pod/v3/users?email=example@example.com,example@example.org&local=true",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #getUsersFromIdList handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "get",
      "path": "/pod/v3/users?uid=1234&local=true",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #importMessages handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "keyManagerToken": "key-manager-token",
        "sessionToken": "session-token",
      },
      "hostname": "agent.example.com",
      "method": "post",
      "path": "/agent/v4/message/import",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #listSignals handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "keyManagerToken": "key-manager-token",
        "sessionToken": "session-token",
      },
      "hostname": "agent.example.com",
      "method": "GET",
      "path": "/agent/v1/signals/list?skip=0&limit=50",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #listUsers handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "get",
      "path": "/pod/v2/admin/user/list?skip=0&limit=1000",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #oboGetAllConnections handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "get",
      "path": "/pod/v1/connection/list?status=all",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #oboGetConnection handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "get",
      "path": "/pod/v1/connection/user/abc123/info",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #oboGetUserIMStreamId handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "post",
      "path": "/pod/v1/im/create",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #promoteUserToOwner handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/room/abc123/membership/promoteOwner",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #registerInterestExtUser handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/user/presence/register",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #rejectConnectionRequest handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/connection/reject",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #removeConnection handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/connection/user/abc123/remove",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #removeMemberFromRoom handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/room/abc123/membership/remove",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #searchRooms handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v3/room/search?skip=0&limit=100",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #searchUsers handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "post",
      "path": "/pod/v1/user/search?local=true&skip=10&limit=100",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #sendConnectionRequest handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v1/connection/create",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #setPresence handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v2/user/presence",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #streamMembers handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "get",
      "path": "/pod/v1/admin/stream/abc123/membership/list?skip=0&limit=1000",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #subscribeSignal handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "keyManagerToken": "key-manager-token",
        "sessionToken": "session-token",
      },
      "hostname": "agent.example.com",
      "method": "POST",
      "path": "/agent/v1/signals/abc123/subscribe?pushed=false",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #suppressMessage handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "post",
      "path": "/pod/v1/admin/messagesuppression/abc123/suppress",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #unsubscribeSignal handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "keyManagerToken": "key-manager-token",
        "sessionToken": "session-token",
      },
      "hostname": "agent.example.com",
      "method": "POST",
      "path": "/agent/v1/signals/abc123/unsubscribe",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #updateRoom handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "sessionToken": "session-token",
      },
      "hostname": "pod.example.com",
      "method": "POST",
      "path": "/pod/v3/room/abc123/update",
      "port": 443,
    },
    [Function],
  ],
]
`;

exports[`SymBotClient #updateSignal handles success 1`] = `
Array [
  Array [
    Object {
      "agent": undefined,
      "headers": Object {
        "Content-Type": "application/json",
        "keyManagerToken": "key-manager-token",
        "sessionToken": "session-token",
      },
      "hostname": "agent.example.com",
      "method": "POST",
      "path": "/agent/v1/signals/abc123/update",
      "port": 443,
    },
    [Function],
  ],
]
`;
